### PR TITLE
refactor: add generic Indexer to E2ETestEnv 

### PR DESCRIPTION
**Pull request by sergeytimoshin**

### CODE REVIEW

File: forester/src/indexer/photon_indexer.rs [hunk 1]
```diff
@@ -1,28 +1,32 @@
...
-use log::info;
+use light_test_utils::rpc::rpc_connection::RpcConnection;
+use log::{debug, info};
 use photon_api::apis::configuration::Configuration;
 use photon_api::models::GetCompressedAccountsByOwnerPostRequestParams;
 use solana_sdk::bs58;
 use std::fmt::Debug;
 
-pub struct PhotonIndexer {
+pub struct PhotonIndexer<const INDEXED_ARRAY_SIZE: usize, R: RpcConnection> {
```
Comment (ananas-block) [inline forester/src/indexer/photon_indexer.rs:11]: Const generics here feel weird to me since the local photon client will not hold an indexed array.

Comment (sergeytimoshin) [reply forester/src/indexer/photon_indexer.rs:11]: Agreed, it's ugly. Test indexer has generic R in one of the methods, that needs to be unified in trait. Here: 
1. https://github.com/Lightprotocol/light-protocol/pull/914/files#diff-f00f25b99320fcd210d228457a61aad543f557fb69cfca88889f455db0d396c2R105
2. https://github.com/Lightprotocol/light-protocol/pull/914/files#diff-deb92dc32bc5ec1000b929e1de32f88bc0f510331eca1f276dc4fe450da7db37R498-R505

Comment (vadorovsky) [reply forester/src/indexer/photon_indexer.rs:11]: I think we should just remove the const generic from indexed array in the first place.

I must've been braindead when introducing it in the first place. It's not used on-chain, so there is absolutely no reason to use arrays. We could probably just use a simple `Vec` there.

Comment (ananas-block) [reply forester/src/indexer/photon_indexer.rs:11]: I think we did use it onchain for a little bit before we had hashset queues

File: merkle-tree/indexed/src/copy.rs [hunk 2]
```diff
@@ -56,13 +55,29 @@ where
         let indexed_changelog_metadata: CyclicBoundedVecMetadata =
             unsafe { read_value_at(bytes, &mut offset) };
 
+        println!("merkle_tree.height: {:?}", merkle_tree.height);
```
Comment (vadorovsky) [inline merkle-tree/indexed/src/copy.rs:58]: 1. Leftover `println!`, pls remove - this one and the others in this file.
2. I guess you added these to make sure that the copy wrapper works correctly. Did you spot anything sketchy?

File: merkle-tree/indexed/src/lib.rs [hunk 3]
```diff
@@ -84,6 +84,16 @@ where
         canopy_depth: usize,
         indexed_changelog_size: usize,
     ) -> usize {
+        println!(
```
Comment (vadorovsky) [inline merkle-tree/indexed/src/lib.rs:87]: Leftover `println!`

**Review by vadorovsky — CHANGES_REQUESTED** [review]

File: merkle-tree/indexed/src/copy.rs (hunk 2 above)
Comment (sergeytimoshin) [reply merkle-tree/indexed/src/copy.rs]: >  Did you spot anything sketchy?

No, lgtm!

**Review by ananas-block — APPROVED** [review]

**Review by vadorovsky — APPROVED** [review]

**Pull request merged**